### PR TITLE
Add Rust support for reading config files

### PR DESCRIPTION
**Pull request by mphuong94**
*Issue #, if available:*

*Description of changes:*
- Adding Rust trait for interface
- Add function to check version and pull appropriate file

---
*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: guard/src/commands/external.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+pub trait Pull {
```
Comment (akshayrane) [inline guard/src/commands/external.rs]: Can we instead create a trait for a generic source, call it `ExternalSource` and make it contain all the methods that a typical source should have? Think from perspective of all new sources that we might add and what will be the common denominator.

File: guard/src/commands/github_pull.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    /// Function to print detail of the instance
+    pub fn to_string(&self) -> String {
+        format!("GitHubSource user({}) repo ({}) file_name({}) access_token({}) version_needed({})",
+                &self.user, &self.repo, &self.file_name, &self.access_token, &self.version_needed)
+    }
+
+    /// Function to read config from TOML
+    fn read_config(&self) ->GitHubSource {
```
Comment (akshayrane) [inline guard/src/commands/github_pull.rs]: this method must be out in a separate struct which will become part of `utils` crate, since we are reading a config and configs could be generic and not specific to Github source.

What we should keep here instead is a constructor that accepts all parameters post sanity check. (looks like you already have  one on line 16.

File: guard/src/commands/github_pull.rs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                output = version;
+                break
+            }
+        }
+        return output;
+    }
+}
+
+/// Function to pull content from GitHub
+impl Pull for GitHubSource {
```
Comment (akshayrane) [inline guard/src/commands/github_pull.rs]: implement all methods that we later decide, they should be inherited from `ExternalSource` trait

File: guard/src/commands/github_pull_test.rs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+use super::super::*;
+use super::*;
+
+#[test]
```
Comment (akshayrane) [inline guard/src/commands/github_pull_test.rs:4]: We would need more unit tests, think of these tests in below categories:

1. check whether methods work as they should with different combinations of inputs passed to them - the tests should show different behaviors being handled, and name of the test should reflect what behavior is expected
2. make tests elementary, they should be clear and concise and should test a very small part of your code such as a method, a constructor and then try to increase coverage by covering wider scenarios such as invocation of a routine involving multiple method calls (this may come after we have already tested the smaller code paths)
3. check for failures, exceptions in case of invalid input, write negative unit tests for these
4. check for boundary conditions, think of scenarios that may exist and are specific to your code - such as
    1. what would happen if credentials passed are incorrect?
    2. what happens in case the repository doesn't exist?
    3. what would happen in case user's API token does not have access to the repo?

**Review by akshayrane — CHANGES_REQUESTED** [review]

File: guard/src/commands/github_pull.rs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+    /// Function to read config from TOML
+    fn read_config(&self) ->GitHubSource {
+        let settings = Config::builder()
+            // file name is "ExternalSourceConfig"
+            .add_source(config::File::with_name("src/ExternalSourceConfig"))
+            .build()
+            .unwrap();
+
+        // put details into a hashmap and create a new instance
```
Comment (akshayrane) [inline guard/src/commands/github_pull.rs]: When we move this logic to a util crate, we should add multiple methods for 
1. reading from config file (parsing logic)
2. retrieving a specific parameter (this will be a wrapper on top of config library's inbuilt methods)
3. doing a sanity check for types of parameters
4. doing a sanity check around values of parameters

**Review by akshayrane — CHANGES_REQUESTED** [review]
